tools/cloud_functions/gcs_event_based_ingest/external_query.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Command Line utility for dry running BigQuery queries that reference
+temporary external tables over data in GCS.
+"""
+import argparse
+import json
+import logging
+import sys
+from typing import List
+
+from google.cloud import bigquery
+from google.cloud import storage
+
+import gcs_ocn_bq_ingest.common.utils  # pylint: disable=import-error
+
+
+def main(args: argparse.Namespace):
+    """main entry point for dry run external CLI."""
+    bq_client: bigquery.Client = bigquery.Client()
+    gcs_client: storage.Client = storage.Client()
+    gsurl = None
+    if args.external_config.startswith("gs://"):
+        gsurl = args.external_config
+        external_config = bigquery.ExternalConfig.from_api_repr(
+            gcs_ocn_bq_ingest.common.utils.read_gcs_file(gcs_client, gsurl))
+    else:
+        with open(args.external_config, 'r') as external_config_file:
+            external_config = bigquery.ExternalConfig.from_api_repr(
+                json.load(external_config_file))
+
+    if (not external_config.source_uris
+            or external_config.source_uris == ["REPLACEME"]):
+        if gsurl:
+            parent_gsurl = "/".join(gsurl.split("/")[:-1])
+            external_config.source_uris = f"{parent_gsurl}/*"
```

Comment:
make the prefix configurable and get GCS URIs of non-empty files non-_config files.